### DISCUSSION TITLE
Eraser usage problem

### DISCUSSION THREAD

**Discussion by Tunanby** [fabricjs/fabric.js#7361] [Q&A]
Whether the eraser function allows you to customize path erasing

**Comment by phoenisx** [comment]
I think this is related to https://github.com/fabricjs/fabric.js/issues/7369.

Eraser does allow erasing paths/custom paths, though, with highly scaled path objects, I am having some troubles, which I am not able to resolve.

**Comment by ShaMan123** [comment]
@Tunanby I came across your question now (hope it is still relevant).
It is possible to customize erasing.
What you need to know is that erasing is actually made of 2 seperate steps (as do all brushes):
1. The drawing/erasing - done on the top context
2. after erasing - mutating `clipPath`.
The second step is easier to customize.
Look into the events fired, especially `erasing:end` on canvas and object level, see [source](https://github.com/fabricjs/fabric.js/blob/master/src/mixins/eraser_brush.mixin.js#L806) and the [erasing demo](http://fabricjs.com/erasing).
If you have more specific question I'm happy to help.
